docs/design-docs/08-package-variant.md

```diff
@@ -640,19 +707,530 @@ understanding, we can see that the controller will perform mutations on the
 downstream package in this order, for both creation and update:
 
 1. Create (via Clone) or Update (via `kpt pkg update` equivalent)
-1. Package Context Injections
-1. Kptfile KRM Function Pipeline Additions/Changes
-1. Config Injection
-1. Kptfile KRM Function Pipeline Execution
+   - This is done by the Porch server, not by the package variant controller
+     directly.
+   - This means that Porch will run the Kptfile pipeline after clone or
+     update.
+1. Package variant controller applies configured mutations
+   - Package Context Injections
+   - Kptfile KRM Function Pipeline Additions/Changes
+   - Config Injection
+1. Package variant controller saves the PackageRevision and
+   PackageRevisionResources.
+   - Porch server executes the Kptfile pipeline
+
+The package variant controller mutations edit resources (including the Kptfile),
+based on the contents of the PackageVariant and the injected in-cluster
+resources, but cannot affect one another. The results of those mutations
+throughout the rest of the package is materialized by the execution of the
+Kptfile pipeline during the save operation.
+
+#### PackageVariant Status
+
+PackageVariant sets the following status conditions:
+ - `Stalled` is set to True if there has been a failure that most likely
+   requires user intervention.
+ - `Ready` is set to True if the last reconciliation successfully produced an
+   up-to-date Draft.
+
+The PackageVariant resource will also contain a `DownstreamTargets` field,
```

Comment:
@natasha41575 is this an accurate description?

Reply:
Yes, that is accurate.